.github/workflows/py-prod-deps-smoke-test.yml

```diff
@@ -0,0 +1,38 @@
+name: Python Prod Deps Smoke Test
+
+on:
+  pull_request:
+    types: [opened]
+  push:
+    branches:
+      - '**'
+
+jobs:
+  python-prod-deps-smoke-test:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.9.7
```

Comment:
Same comment as the one here: https://github.com/streamlit/streamlit/pull/5202#discussion_r950617517